compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -75,6 +75,30 @@ pub fn obligations<'tcx>(
     Some(result)
 }
 
+pub fn unnormalized_obligations<'tcx>(
+    infcx: &InferCtxt<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    arg: GenericArg<'tcx>,
+) -> Option<Vec<traits::PredicateObligation<'tcx>>> {
+    if let ty::GenericArgKind::Lifetime(..) = arg.unpack() {
+        return Some(vec![]);
+    }
+
+    assert_eq!(infcx.shallow_resolve(arg), arg);
```

Comment:
could make this `debug_assert_eq!(infcx.resolve_vars_if_possible(arg), arg);` as it currently doesn't catch everything.